### PR TITLE
refactor: Refactor some attributes on the WebClientSocket for readability

### PR DESCRIPTION
**Pull request by lilyyy411**
## About

This pull request is about issue #751, which proposes some refactoring related to the `interactions.api.gateway` package

## Checklist

- [x] I've ran `pre-commit` to format and lint the change(s) made.
- [x] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [x] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [x] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [x] New feature/enhancement
  - [ ] Bugfix

### CODE REVIEW

File: interactions/api/gateway/client.py [hunk 1]
```diff
@@ -507,12 +505,10 @@ def __option_type_context(self, context: object, type: int) -> dict:
     async def restart(self):
         await self.__restart()
 
-    @property
-    async def __receive_packet_stream(self) -> Optional[Dict[str, Any]]:
+    async def __get_next_packet(self) -> Union[Dict[str, Any], WSMessage, None]:
```
Comment (sillyElvy) [inline interactions/api/gateway/client.py:508]: It was meant to be a property, please keep it as that

File: interactions/api/gateway/client.pyi [hunk 2]
```diff
@@ -60,13 +60,12 @@ class WebSocketClient:
...
-    def __contextualize(self, data: dict) -> object: ...
+    def __contextualize(self, data: dict) -> Any: ...
     def __sub_command_context(
         self, data: Union[dict, Option], _context: Optional[object] = MISSING
     ) -> Union[Tuple[str], dict]: ...
-    def __option_type_context(self, context: object, type: int) -> dict: ...
-    @property
-    async def __receive_packet_stream(self) -> Optional[Dict[str, Any]]: ...
+    def __option_type_context(self, context: Any, type: int) -> dict: ...
+    async def __get_next_packet(self) -> Union[Dict[str, Any], WSMessage, None]: ...
```
Comment (sillyElvy) [inline interactions/api/gateway/client.pyi:68]: It's a property

File: interactions/api/gateway/client.pyi [hunk 3]
```diff
@@ -6,9 +6,8 @@ from asyncio import (
 from logging import Logger
 from typing import Any, Dict, List, Optional, Tuple, Union, Iterable
 
-from aiohttp import ClientWebSocketResponse
+from aiohttp import ClientWebSocketResponse, WSMessage
 
-from .heartbeat import _Heartbeat
```
Comment (sillyElvy) [inline interactions/api/gateway/client.pyi:11]: I'm against the removal of the heartbeat class, please revert this changes

File: interactions/api/gateway/client.pyi [hunk 4]
```diff
@@ -60,13 +60,12 @@ class WebSocketClient:
...
     def _dispatch_event(self, event: str, data: dict) -> None: ...
-    def __contextualize(self, data: dict) -> object: ...
+    def __contextualize(self, data: dict) -> Any: ...
     def __sub_command_context(
         self, data: Union[dict, Option], _context: Optional[object] = MISSING
     ) -> Union[Tuple[str], dict]: ...
-    def __option_type_context(self, context: object, type: int) -> dict: ...
-    @property
-    async def __receive_packet_stream(self) -> Optional[Dict[str, Any]]: ...
+    def __option_type_context(self, context: Any, type: int) -> dict: ...
```
Comment (sillyElvy) [inline interactions/api/gateway/client.pyi:67]: Why are you changing the typehint?

**Review by sillyElvy — CHANGES_REQUESTED** [review]

File: interactions/api/gateway/client.pyi (hunk 3 above)
Comment (FayeDel) [reply interactions/api/gateway/client.pyi:11]: ??
What's the explanation behind this comment?

Comment (sillyElvy) [reply interactions/api/gateway/client.pyi:11]: I don't see the reason for a removal, imo it is an unnecessary change

Comment (AstreaTSS) [reply interactions/api/gateway/client.pyi:11]: I mean... removing an extra, unnecessary class makes sense if possible, and it seems like all the class did was add overhead to the gateway.

Comment (sillyElvy) [reply interactions/api/gateway/client.pyi:11]: If you're both fine with removing it we can do it

File: interactions/api/gateway/client.py (hunk 1 above)
Comment (AstreaTSS) [reply interactions/api/gateway/client.py:508]: A property being asynchronous is... odd, to be fair. I can see why it was changed to be a function.

I'm not *against* keeping it as a property per se, but what was the reason for making it a property?

Comment (sillyElvy) [reply interactions/api/gateway/client.py:508]: uh.. @jameswalston? ..
